implementations/rust/ockam/ockam_vault_test_suite/src/signer_impl.rs

```diff
@@ -14,7 +15,9 @@ pub fn sign(vault: &mut (impl Signer + Verifier + SecretVault)) {
     let res = vault.sign(&secret, b"hello world!");
     assert!(res.is_ok());
     let pubkey = vault.secret_public_key_get(&secret).unwrap();
+    //*array_ref![digest, 0, 32]
     let signature = res.unwrap();
-    let res = vault.verify(&signature, &pubkey, b"hello world!").unwrap();
+    let signature_array = *array_ref![signature, 0, 64];
```

Comment:
Why do we need this? Is there now difference in types that sign produces and verify accepts?

Reply:
Yeah. I have mixed it but now, since Signature type is there as you've suggested it should be fine for testing. The same on the sign() side and verify() side.